### PR TITLE
Fixes #17005 - more strict debug password filter

### PR DESCRIPTION
**Pull request by lzap**
We have reports on missed password entries in logs and config files, see the
ticket for the complete list. This patch covers answer files (default files)
and also certificate key files.

### CODE REVIEW

File: script/foreman-debug [hunk 1]
```diff
@@ -39,20 +39,20 @@ Please note that the rsync transmission is UNENCRYPTED.
 UPLOADUSAGE
 }
 
-# filter for patterns like password=XYZ or secret: abc!@#$123
+# Filter for patterns like password=XYZ, -storepass XYZ or secret: XYZ
 FILTER_WORDS=(
 password
 PASSWORD
 default_password
+key
```
Comment (domcleal) [inline script/foreman-debug]: This causes paths in installer answer files or Foreman's own settings.yaml (e.g. ssl_priv_key) to be filtered out, e.g. `client_ssl_key`, is that intentional?

File: script/foreman-debug [hunk 2]
```diff
@@ -39,20 +39,20 @@ Please note that the rsync transmission is UNENCRYPTED.
...
 token
-api_token
-oauth_secret
 keystorePass
 truststorePass
+storepass
 )
 FILTER_WORDS_STR=$(IFS=$'|'; echo "${FILTER_WORDS[*]}")
-FILTER="s/($FILTER_WORDS_STR)(\s*[:=]\s*)\S+/\1\2\*\*\*\*\*/g"
+FILTER="s/($FILTER_WORDS_STR)(\"?\s*[:= ]\s*)\S+/\1\2+FILTERED+/g"
```
Comment (domcleal) [inline script/foreman-debug]: No need for the extra space here? \s\* is either side of it.

File: script/foreman-debug [hunk 3]
```diff
@@ -287,7 +287,9 @@ add_cmd "gem list" "gem_list"
 add_cmd "scl enable $SCLNAME 'gem list'" "gem_list_scl"
 add_cmd "bundle --local --gemfile=/usr/share/foreman/Gemfile" "bundle_list"
 add_cmd "facter" "facts"
-add_files /etc/foreman/* /var/log/foreman/*.log*
+# exclude *key.pem files and encryption_key.rb
+add_files /etc/foreman/*.{yml,yaml,conf} /etc/foreman/*{cert,ca}.pem /etc/foreman/plugins/
```
Comment (domcleal) [inline script/foreman-debug]: 1. No need for *{cert,ca}.pem here I think, that depends on the user's configuration? If it's from Katello, please add it to their debug scripts.
2. /etc/foreman/plugins/ is missing a filename, so it's not picking up any files. *.yaml might be most precise (that's all that's loaded anyway).

**Review by domcleal — CHANGES_REQUESTED** [review]
There are also foreman-proxy related bits of the bug report, please file a new ticket against the Smart Proxy and fix them in its foreman-debug script too.

File: script/foreman-debug (hunk 1 above)
Comment (lzap) [reply script/foreman-debug]: Yes, the problem with current approach mentioning only specific strings was not working apparently, this patch tighten the filtering causing a lot more false alerts which should be safer. If you don't like, I can simply name them.

File: script/foreman-debug (hunk 2 above)
Comment (lzap) [reply script/foreman-debug]: Yeah thanks.

File: script/foreman-debug (hunk 3 above)
Comment (lzap) [reply script/foreman-debug]: Removed katello related certs, added `yaml` extension.

**Comment by lzap** [comment]
Resolved conflicts.

**Comment by lzap** [comment]
Here is an example session, Foreman 1.13:

```
/tmp/debug/var/log/foreman/production.log:2016-10-25 02:41:08 e16b3a18 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "login"=>{"login"=>"admin", "password"=+FILTERED+ "commit"=>"Login"}
/tmp/debug/var/log/foreman/production.log:2016-10-25 02:41:16 45c825f8 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "host"=>{"name"=>"gwen-navappo", "hostgroup_id"=>"1", "managed"=>"true", "progress_report_id"=>"[FILTERED]", "type"=>"Host::Managed", "interfaces_attributes"=>{"0"=>{"_destroy"=>"0", "type"=>"Nic::Managed", "mac"=>"", "identifier"=>"", "name"=>"gwen-navappo", "domain_id"=>"", "subnet_id"=>"", "ip"=>"", "ip6"=>"", "managed"=>"1", "primary"=>"1", "provision"=>"1", "virtual"=>"0", "tag"=>"", "attached_to"=>""}}, "architecture_id"=>"", "provision_method"=>"build", "build"=>"1", "disk"=>"", "root_pass"=>"[FILTERED]", "is_owned_by"=>"3-Users", "enabled"=>"1", "model_id"=>"", "comment"=>"", "overwrite"=>"false"}, "fakepassword"=+FILTERED+ "bare_metal_capabilities"=>"build"}
/tmp/debug/var/log/foreman/production.log:2016-10-25 02:41:17 45c825f8 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "host"=>{"name"=>"gwen-navappo", "hostgroup_id"=>"1", "compute_resource_id"=>"", "puppetclass_ids"=>[""], "managed"=>"true", "progress_report_id"=>"[FILTERED]", "type"=>"Host::Managed", "interfaces_attributes"=>{"0"=>{"_destroy"=>"0", "type"=>"Nic::Managed", "mac"=>"", "identifier"=>"", "name"=>"gwen-navappo", "domain_id"=>"2", "subnet_id"=>"1", "ip"=>"", "ip6"=>"", "managed"=>"1", "primary"=>"1", "provision"=>"1", "virtual"=>"0", "tag"=>"", "attached_to"=>""}}, "architecture_id"=>"1", "operatingsystem_id"=>"1", "provision_method"=>"build", "build"=>"1", "medium_id"=>"1", "ptable_id"=>"80", "pxe_loader"=>"Grub2 UEFI", "disk"=>"", "root_pass"=>"[FILTERED]", "is_owned_by"=>"3-Users", "enabled"=>"1", "model_id"=>"", "comment"=>"", "overwrite"=>"false"}, "fakepassword"=+FILTERED+ "bare_metal_capabilities"=>"build"}
/tmp/debug/var/log/foreman/production.log:2016-10-25 02:41:51 45c825f8 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "host"=>{"name"=>"gwen-navappo", "hostgroup_id"=>"1", "compute_resource_id"=>"", "puppetclass_ids"=>[""], "managed"=>"true", "progress_report_id"=>"[FILTERED]", "type"=>"Host::Managed", "interfaces_attributes"=>{"0"=>{"_destroy"=>"0", "type"=>"Nic::Managed", "mac"=>"52:54:00:60:d0:fe", "identifier"=>"", "name"=>"gwen-navappo", "domain_id"=>"2", "subnet_id"=>"1", "ip"=>"192.168.222.100", "ip6"=>"", "managed"=>"1", "primary"=>"1", "provision"=>"1", "virtual"=>"0", "tag"=>"", "attached_to"=>""}, "1477377700813"=>{"_destroy"=>"0", "type"=>"Nic::Managed", "mac"=>"52:54:00:60:d0:fe", "identifier"=>"", "name"=>"", "domain_id"=>"", "subnet_id"=>"1", "ip"=>"192.168.222.101", "ip6"=>"", "managed"=>"1", "primary"=>"0", "provision"=>"0", "virtual"=>"0", "tag"=>"", "attached_to"=>""}}, "architecture_id"=>"1", "operatingsystem_id"=>"1", "provision_method"=>"build", "build"=>"1", "medium_id"=>"1", "ptable_id"=>"80", "pxe_loader"=>"Grub2 UEFI", "disk"=>"", "root_pass"=>"[FILTERED]", "is_owned_by"=>"3-Users", "enabled"=>"1", "model_id"=>"", "comment"=>"", "overwrite"=>"false"}, "bare_metal_capabilities"=>"build"}
/tmp/debug/var/log/foreman/production.log:2016-10-25 09:30:42 9130b1f7 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "login"=>{"login"=>"admin", "password"=+FILTERED+ "commit"=>"Login"}
/tmp/debug/var/log/foreman/production.log:2016-10-25 10:07:11 95be5624 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "host"=>{"name"=>"joni-weller", "hostgroup_id"=>"", "compute_resource_id"=>"", "environment_id"=>"", "puppet_ca_proxy_id"=>"", "puppet_proxy_id"=>"", "managed"=>"true", "progress_report_id"=>"[FILTERED]", "type"=>"Host::Managed", "interfaces_attributes"=>{"0"=>{"_destroy"=>"0", "type"=>"Nic::Managed", "mac"=>"", "identifier"=>"", "name"=>"joni-weller", "domain_id"=>"", "subnet_id"=>"", "ip"=>"", "ip6"=>"", "managed"=>"1", "primary"=>"1", "provision"=>"1", "virtual"=>"0", "tag"=>"", "attached_to"=>""}}, "architecture_id"=>"", "provision_method"=>"build", "build"=>"1", "disk"=>"", "root_pass"=>"[FILTERED]", "is_owned_by"=>"3-Users", "enabled"=>"1", "model_id"=>"", "comment"=>"", "overwrite"=>"false"}, "bare_metal_capabilities"=>"build"}
/tmp/debug/var/log/foreman/production.log:2016-10-25 10:07:25 95be5624 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "host"=>{"name"=>"joni-weller", "hostgroup_id"=>"1", "managed"=>"true", "progress_report_id"=>"[FILTERED]", "type"=>"Host::Managed", "interfaces_attributes"=>{"0"=>{"_destroy"=>"0", "type"=>"Nic::Managed", "mac"=>"52:54:00:c6:a9:df", "identifier"=>"", "name"=>"joni-weller", "domain_id"=>"", "subnet_id"=>"", "ip"=>"", "ip6"=>"", "managed"=>"1", "primary"=>"1", "provision"=>"1", "virtual"=>"0", "tag"=>"", "attached_to"=>""}}, "architecture_id"=>"", "provision_method"=>"build", "build"=>"1", "disk"=>"", "root_pass"=>"[FILTERED]", "is_owned_by"=>"3-Users", "enabled"=>"1", "model_id"=>"", "comment"=>"", "overwrite"=>"false"}, "fakepassword"=+FILTERED+ "bare_metal_capabilities"=>"build"}
/tmp/debug/var/log/foreman/production.log:2016-10-25 10:07:25 95be5624 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "host"=>{"name"=>"joni-weller", "hostgroup_id"=>"1", "compute_resource_id"=>"", "puppetclass_ids"=>[""], "managed"=>"true", "progress_report_id"=>"[FILTERED]", "type"=>"Host::Managed", "interfaces_attributes"=>{"0"=>{"_destroy"=>"0", "type"=>"Nic::Managed", "mac"=>"52:54:00:c6:a9:df", "identifier"=>"", "name"=>"joni-weller", "domain_id"=>"2", "subnet_id"=>"1", "ip"=>"", "ip6"=>"", "managed"=>"1", "primary"=>"1", "provision"=>"1", "virtual"=>"0", "tag"=>"", "attached_to"=>""}}, "architecture_id"=>"1", "operatingsystem_id"=>"1", "provision_method"=>"build", "build"=>"1", "medium_id"=>"1", "ptable_id"=>"80", "pxe_loader"=>"Grub2 UEFI", "disk"=>"", "root_pass"=>"[FILTERED]", "is_owned_by"=>"3-Users", "enabled"=>"1", "model_id"=>"", "comment"=>"", "overwrite"=>"false"}, "fakepassword"=+FILTERED+ "bare_metal_capabilities"=>"build"}
/tmp/debug/var/log/foreman/production.log:2016-10-25 10:07:26 95be5624 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "host"=>{"name"=>"joni-weller", "hostgroup_id"=>"1", "compute_resource_id"=>"", "puppetclass_ids"=>[""], "managed"=>"true", "progress_report_id"=>"[FILTERED]", "type"=>"Host::Managed", "interfaces_attributes"=>{"0"=>{"_destroy"=>"0", "type"=>"Nic::Managed", "mac"=>"52:54:00:c6:a9:df", "identifier"=>"", "name"=>"joni-weller", "domain_id"=>"2", "subnet_id"=>"1", "ip6"=>"", "managed"=>"1", "primary"=>"1", "provision"=>"1", "virtual"=>"0", "tag"=>"", "attached_to"=>""}}, "architecture_id"=>"1", "operatingsystem_id"=>"1", "provision_method"=>"build", "build"=>"1", "medium_id"=>"1", "ptable_id"=>"80", "pxe_loader"=>"Grub2 UEFI", "disk"=>"", "root_pass"=>"[FILTERED]", "is_owned_by"=>"3-Users", "enabled"=>"1", "model_id"=>"", "comment"=>"", "overwrite"=>"false"}, "bare_metal_capabilities"=>"build"}
/tmp/debug/var/log/foreman/production.log:2016-10-25 10:07:38 95be5624 [app] [I]   Parameters: {"authenticity_token"=+FILTERED+ "id"=>"joni-weller.nested.lan"}
/tmp/debug/var/log/foreman/production.log:2016-11-01 10:50:07 702e8084 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "login"=>{"login"=>"admin", "password"=+FILTERED+ "commit"=>"Login"}
/tmp/debug/var/log/foreman/production.log:2016-11-02 08:28:53 ba442256 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "login"=>{"login"=>"admin", "password"=+FILTERED+ "commit"=>"Login"}
/tmp/debug/var/log/foreman/production.log:2016-11-02 09:08:26 1fae37c2 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "changed"=>{"new"=>{"production"=>"{\"stdlib::stages\":{\"new\":{}},\"stdlib\":{\"new\":{}},\"ntp::service\":{\"new\":{}},\"ntp::config\":{\"new\":{}},\"ntp::install\":{\"new\":{}}}"}}, "commit"=>"Update"}
/tmp/debug/var/log/foreman/production.log:2016-11-02 10:49:16 b65fc76f [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "login"=>{"login"=>"admin", "password"=+FILTERED+ "commit"=>"Login"}
/tmp/debug/var/log/foreman/production.log:2016-11-03 09:23:00 a801718a [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "login"=>{"login"=>"admin", "password"=+FILTERED+ "commit"=>"Login"}
/tmp/debug/var/log/foreman/production.log:2016-11-03 10:45:04 20f0504a [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "login"=>{"login"=>"admin", "password"=+FILTERED+ "commit"=>"Login"}
/tmp/debug/var/log/foreman/production.log:2016-11-07 11:07:38 da3f2875 [app] [I]   Parameters: {"utf8"=>"✓", "authenticity_token"=+FILTERED+ "login"=>{"login"=>"admin", "password"=+FILTERED+ "commit"=>"Login"}
/tmp/debug/var/log/foreman/production.log:2016-11-07 11:07:50 21346360 [app] [I]   Parameters: {"authenticity_token"=+FILTERED+ "id"=>"mac5254005c0928"}
/tmp/debug/etc/foreman/settings.yaml::oauth_consumer_key: +FILTERED+
/tmp/debug/etc/foreman/settings.yaml::oauth_consumer_secret: +FILTERED+
/tmp/debug/etc/foreman/settings.yaml::websockets_ssl_key: +FILTERED+
/tmp/debug/etc/foreman/settings.yaml::ssl_priv_key: +FILTERED+
/tmp/debug/etc/foreman/database.yml:  password: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman-answers.yaml:    db_password: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman-answers.yaml:    server_ssl_key: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman-answers.yaml:    client_ssl_key: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman-answers.yaml:    oauth_consumer_key: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman-answers.yaml:    oauth_consumer_secret: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman-answers.yaml:    admin_password: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman-answers.yaml:    websockets_ssl_key: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman-answers.yaml:    ssl_key: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman-answers.yaml:    puppet_ssl_key: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman-answers.yaml:    oauth_consumer_key: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman-answers.yaml:    oauth_consumer_secret: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman.yaml::password: +FILTERED+
/tmp/debug/etc/foreman-installer/scenarios.d/foreman.yaml::password: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:19:26 main]  /Stage[main]/Foreman::Repos::Extra/Yumrepo[epel]/gpgkey: +FILTERED+ changed 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7' to 'https://fedoraproject.org/static/352C64E5.txt'
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:21:58 main] -  #:password: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:21:58 main] +    hostprivkey = +FILTERED+ { mode = 640 }
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:02 main] -#:dns_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:02 main] +:dns_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:02 main]  #:key_secret: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:02 main] -#:puppet_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:02 main] +:puppet_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:03 main] -#:puppet_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:03 main] +:puppet_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:03 main] -#:ssl_private_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:03 main] +:ssl_private_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:03 main]  #:foreman_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:11 main] +  password: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:12 main] +:oauth_consumer_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:12 main] +:oauth_consumer_secret: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:12 main] +:websockets_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:12 main] +:ssl_priv_key: +FILTERED+
/tmp/debug/etc/libvirt/qemu.conf:#vnc_password = +FILTERED+
/tmp/debug/etc/libvirt/qemu.conf:#spice_password = +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:19:26 main]  /Stage[main]/Foreman::Repos::Extra/Yumrepo[epel]/gpgkey: +FILTERED+ changed 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7' to 'https://fedoraproject.org/static/352C64E5.txt'
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:21:58 main] -  #:password: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:21:58 main] +    hostprivkey = +FILTERED+ { mode = 640 }
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:02 main] -#:dns_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:02 main] +:dns_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:02 main]  #:key_secret: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:02 main] -#:puppet_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:02 main] +:puppet_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:03 main] -#:puppet_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:03 main] +:puppet_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:03 main] -#:ssl_private_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:03 main] +:ssl_private_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:03 main]  #:foreman_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:11 main] +  password: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:12 main] +:oauth_consumer_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:12 main] +:oauth_consumer_secret: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:12 main] +:websockets_ssl_key: +FILTERED+
/tmp/debug/var/log/foreman-installer/foreman.3.log:[ WARN 2016-10-24 03:22:12 main] +:ssl_priv_key: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.yml::ssl_private_key: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.yml:#:foreman_ssl_key: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.d/puppet_proxy_legacy.yml::puppet_ssl_key: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.d/puppet_proxy_legacy.yml::puppet_ssl_key: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.d/puppet_proxy_puppet_api.yml::puppet_ssl_key: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.d/puppet_proxy_puppet_api.yml::puppet_ssl_key: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.d/dhcp_isc.yml:#:key_secret: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.d/dhcp_isc.yml:#:key_secret: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.d/dns_nsupdate.yml::dns_key: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.d/dns_nsupdate.yml::dns_key: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.d/abrt.yml:#:server_ssl_key: +FILTERED+
/tmp/debug/etc/foreman-proxy/settings.d/abrt.yml:#:server_ssl_key: +FILTERED+
/tmp/debug/etc/puppet/puppet.conf:    hostprivkey = +FILTERED+ { mode = 640 }
```

File: script/foreman-debug (hunk 1 above)
Comment (domcleal) [reply script/foreman-debug]: What strings are you talking about specifically? The ticket mentions "oauth_consumer_key" - if that's the only one (and I'm unsure it's a secret, as there is a separate secret), then it's probably worth naming.

**Comment by lzap** [comment]
Note to myself: Additional string in installer log:

```
> [ INFO 2015-07-20 10:19:51 main] +    "proxy_username": "my_username",
> [ INFO 2015-07-20 10:19:51 main] +    "proxy_password": "SomeSecretPassword"
```

**Comment by lzap** [comment]
Ok added the full `oauth_consumer_key`. Summary, this should be filtered out:

```
  "capsule::params::pulp_admin_password": XXXXXXXXXXXXXXXXXXXXXXXXXXXX
  "::foreman::params::db_password": XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
  "::foreman::params::oauth_consumer_key": XXXXXXXXXXXXXXXXXXXXXXXXXXX
  "::foreman::params::oauth_consumer_secret": XXXXXXXXXXXXXXXXXXXXXXXXXXXXX
  "::foreman::params::admin_password": XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
  "foreman_proxy::params::oauth_consumer_key": XXXXXXXXXXXXXXXXXXXXXXXXXX
  "foreman_proxy::params::oauth_consumer_secret": XXXXXXXXXXXXXXXXXXXXXXXXXX
  "katello::params::oauth_secret": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXX" 
  "katello::params::post_sync_token": XXXXXXXXXXXXXXXXXXXXXXXXXXX
 [ INFO 2015-07-20 10:19:51 main] +    "proxy_password": "SomeSecretPassword"
```

**Review by domcleal — APPROVED** [review]

**Comment by domcleal** [comment]
Merged, thanks @lzap. Please complete the security process.

**Pull request merged**